packages/components/src/components/expansion-panel/expansion-panel.jsx

```diff
@@ -3,34 +3,30 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import Icon from '../icon';
 
-class ExpansionPanel extends React.Component {
-    state = {
-        open_ids: [],
-        is_open: false,
-    };
+const ExpansionPanel = ({ message, onResize }) => {
+    const [open_ids, setOpenIds] = React.useState([]);
+    const [is_open, setIsOpen] = React.useState(false);
+
+    React.useEffect(() => {
+        if (typeof onResize === 'function') {
+            onResize();
+        }
+    }, [is_open]);
 
-    onClick = () => {
+    const onClick = () => {
         // close if clicking the expansion panel that's open, otherwise open the new one
-        this.setState({ is_open: !this.state.is_open });
+        setIsOpen(!is_open);
     };
 
-    onArrayItemClick = id => {
-        this.setState(state => {
-            if (state.open_ids.includes(id)) {
-                const open_ids = state.open_ids.filter(open_id => open_id !== id);
-                return {
-                    ...state,
-                    open_ids,
-                };
-            }
-            return {
-                ...state,
-                open_ids: [...state.open_ids, id],
-            };
-        });
+    const onArrayItemClick = id => {
+        if (open_ids.includes(id)) {
+            setOpenIds(open_ids.filter(open_id => open_id !== id));
+        } else {
+            setOpenIds([...open_ids, id]);
+        }
     };
 
-    renderArrayContent = array => {
+    const renderArrayContent = array => {
```

Comment:
Since this is a component we can name is as such for clarity, e.g. `ArrayRenderer`, also we should probably move it to outside the `ExpansionPanel` so that it doesn't keep unmounting and remounting if the parent re-renders. In general, we should try to avoid creating components inside of other components.

_**Disclaimer**: I know you didn't write the original code._

Reply:
Nice idea, I moved it to a separate component